.github/workflows/windows.yml

```diff
@@ -17,16 +17,45 @@ jobs:
             arch: x64
             max_warnings: 2170
     steps:
-      - uses:  actions/checkout@v2
-      - name:  Install packages
+      - name:  Backup existing vcpkg installation
+        shell: pwsh
+        run: |
+          mv c:\vcpkg c:\vcpkg-bak
+          md c:\vcpkg -ea 0
+
+      - name: Generate vcpkg cache key
+        id:    prep-vcpkg
+        shell: bash
+        run: |
+          echo "::set-output name=year_and_week::$(date '+%Y%W')"
+
+      - name: Cache vcpkg
+        uses: actions/cache@v2
+        id:   cache-vcpkg
+        with:
+          path: c:\vcpkg
+          key: vcpkg-${{ matrix.conf.arch }}-${{ steps.prep-vcpkg.outputs.year_and_week }}-1
+
+      - name:  Install new packages using vcpkg
+        if:    steps.cache-vcpkg.outputs.cache-hit != 'true'
```

Comment:
I don't get it - isn't cache created in `Cache vcpkg` step missing the packages we want installed?

Reply:
The `Cache vcpkg` step is another terribly documented aspect of GiTHub workflows.

The `uses: actions/cache@v2` action is a "dual role" function that will either find an extract and existing cache or create the cache (with the specified key) _at the end of the job - provided it's successful_ 


``` yaml
      - name: Cache vcpkg
        uses: actions/cache@v2
        id:   cache-vcpkg
        with:
          path: c:\vcpkg
          key: vcpkg-${{ matrix.conf.arch }}-${{ steps.prep-vcpkg.outputs.year_and_week }}-1
```

So coming out of this step - we either have a fully populated vcpkg contained all of dosbox-staging's already compiled depenedencies.. or we have nothing :-)  Actually, failures can also occur mid-way through extraction, which is why I protect against those using the directory backup and rename.

The `if:` below detects the failed or missing cache scenario, and then carries on how we normally install packages (~10 minutes later).

``` yaml
      - name:  Install new packages using vcpkg
        if:    steps.cache-vcpkg.outputs.cache-hit != 'true'

```